scripts/superdesk-search/react.js

```diff
@@ -466,7 +495,17 @@
 
                         render: function() {
                             var item = this.props.item.archive_item || this.props.item;
-                            var highlights = item.highlights || [];
+
+                            var highlights = [];
+                            if (isCheckAllowed(this.props.item)) {
+                                if (this.props.item.archive_item && this.props.item.archive_item.highlights) {
```

Comment:
seems like check for `this.props.item.archive_item.highlights.length` should be also part of this, otherwise it's a bit confusing with `? :` right after